packages/preview-middleware-client/src/adp/quick-actions/fe-v2/lr-enable-semantic-date-range-filter-bar.ts

```diff
@@ -0,0 +1,80 @@
+import FlexCommand from 'sap/ui/rta/command/FlexCommand';
+import CommandFactory from 'sap/ui/rta/command/CommandFactory';
+import type FilterBar from 'sap/ui/comp/filterbar/FilterBar';
+
+import { QuickActionContext, SimpleQuickActionDefinition } from '../../../cpe/quick-actions/quick-action-definition';
+import { pageHasControlId } from '../../../cpe/quick-actions/utils';
+import { getControlById } from '../../../utils/core';
+import { SimpleQuickActionDefinitionBase } from '../simple-quick-action-base';
+import SmartFilterBar from 'sap/ui/comp/smartfilterbar/SmartFilterBar';
+
+export const ENABLE_SEMANTIC_DATE_RANGE_FILTER_BAR = 'enable-semantic-daterange-filterbar';
+const CONTROL_TYPE = 'sap.ui.comp.smartfilterbar.SmartFilterBar';
+
+/**
+ * Quick Action for toggling the visibility of "semantic date range" for filterbar fields.
+ */
+export class ToggleSemanticDateRangeFilterBar
+    extends SimpleQuickActionDefinitionBase
+    implements SimpleQuickActionDefinition
+{
+    constructor(context: QuickActionContext) {
+        super(ENABLE_SEMANTIC_DATE_RANGE_FILTER_BAR, [], '', context);
+    }
+    readonly forceRefreshAfterExecution = true;
+    private isUseDateRangeTypeEnabled = false;
+
+    initialize(): void {
+        const controls = this.context.controlIndex[CONTROL_TYPE] ?? [];
+        for (const control of controls) {
+            const isActionApplicable = pageHasControlId(this.context.view, control.controlId);
+            const modifiedControl = getControlById<FilterBar>(control.controlId);
+            if (isActionApplicable && modifiedControl) {
+                this.isUseDateRangeTypeEnabled = modifiedControl.getProperty('useDateRangeType');
+                this.control = modifiedControl;
+            }
+        }
+    }
+
+    protected get textKey() {
+        return this.isUseDateRangeTypeEnabled
+            ? 'V2_QUICK_ACTION_LR_DISABLE_SEMANTIC_DATE_RANGE_FILTER_BAR'
+            : 'V2_QUICK_ACTION_LR_ENABLE_SEMANTIC_DATE_RANGE_FILTER_BAR';
+    }
+
+    async execute(): Promise<FlexCommand[]> {
+        if (this.control) {
```

Comment:
May be action should be disabled if this.control is undefined?
Otherwise execution will just do nothing and confuse the user.

Reply:
this is handled in the [base class](https://github.com/SAP/open-ux-tools/blob/main/packages/preview-middleware-client/src/adp/quick-actions/simple-quick-action-base.ts#L45),

Reply:
done.